### PR TITLE
Add php version support

### PR DESCRIPTION
**Pull request by stanstst**

### CODE REVIEW

File: classes/Kohana/Jam/Validator/Rule/Count.php [hunk 1]
```diff
@@ -20,7 +20,9 @@ class Kohana_Jam_Validator_Rule_Count extends Jam_Validator_Rule {
 
 	public function validate(Jam_Validated $model, $attribute, $value)
 	{
-		$count = count($value);
+		// Since PHP 7.2 method count() parameter must be an array or an object that implements Countable
+		// https://www.php.net/manual/en/function.count.php#124263
+		$count = count((array) $value);
```
Comment (hkdobrev) [inline classes/Kohana/Jam/Validator/Rule/Count.php]: Ooh, this one could have been a bad one if there were no tests. Usually, casting a random value of an unknown type to an array like that could also lead to problems, but my quick test here looks ok: https://3v4l.org/X6D7n

File: composer.json [hunk 2]
```diff
@@ -13,25 +13,29 @@
...
 		"kohana/image": "^3.3.0"
 	},
 	"require-dev": {
-		"phpunit/phpunit-dom-assertions": "^0.1",
-		"symfony/dom-crawler": ">=2.3",
-		"symfony/css-selector": ">=2.3",
 		"openbuildings/kohana-test-bootstrap": "^0.2",
-		"phpunit/phpunit": "~4.0.0",
-		"openbuildings/jam-monetary": "^0.1"
+		"openbuildings/jam-monetary": "^0.1",
```
Comment (hkdobrev) [inline composer.json]: Shouldn't this package be updated with a newer version?

Comment (TrayanIvanov) [reply composer.json]: Yes, updated.

File: classes/Kohana/Jam/Validator/Rule/Count.php (hunk 1 above)
Comment (TrayanIvanov) [reply classes/Kohana/Jam/Validator/Rule/Count.php]: I was not able to discover a better approach. Do you think this is ok or can you suggest any change?

Comment (hkdobrev) [reply classes/Kohana/Jam/Validator/Rule/Count.php]: No, I had doubts, but that's why I've checked and it looks fine for its purposes.

File: tests/tests/DeepTest.php [hunk 3]
```diff
@@ -32,104 +32,5 @@ public function test_hasone_deep()
 		$this->assertInstanceOf('Model_Test_Post', $author->test_post);
 		$this->assertEquals(3, $author->test_post->id());
 		$this->assertEquals('changed post', $author->test_post->name());
-
-
-	}
-
-	public function test_association_create()
```
Comment (hkdobrev) [inline tests/tests/DeepTest.php]: Is this test moved somewhere? It's quite essential.

Comment (hkdobrev) [reply tests/tests/DeepTest.php]: Oh, I saw it was skipped. I guess we can just enable it and remove anything to do with uploads.

File: tests/classes/Trait/DomSearch.php [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+		if (!empty($content)) {
+			$crawler = $crawler->reduce(
+				function (Crawler $node) use ($content) {
+					if ($content === '') {
+						return $node->text() === '';
+					}
+
+
+					return (bool)preg_match('/' . $content . '/i', $node->text());
```
Comment (hkdobrev) [inline tests/classes/Trait/DomSearch.php]:
```suggestion
					return (bool) preg_match('/' . $content . '/i', $node->text());
```

File: tests/tests/DeepTest.php (hunk 3 above)
Comment (TrayanIvanov) [reply tests/tests/DeepTest.php]: Ok, it's returned.

File: tests/classes/Trait/DomSearch.php (hunk 4 above)
Comment (TrayanIvanov) [reply tests/classes/Trait/DomSearch.php]: Done.

**Pull request merged**